lib/sentry/config.ex

```diff
@@ -253,30 +253,30 @@ defmodule Sentry.Config do
       The maximum number of attempts to send an event to Sentry.
       """
     ],
-    hackney_opts: [
```

Comment:
We need to keep `hackney_opts` and deprecate them if we want this to not be a breaking change. NimbleOptions supports deprecating options, check out the docs for that.

This also applies to the options below.

Reply:
Does this mean we need to keep the option:

``` elixir
    client: [
      type: :atom,
      type_doc: "`t:module/0`",
      default: Sentry.HackneyClient,
      doc: """
      """
    ]
```

and the module HackneyClient??